coffee/expression.py

```diff
@@ -39,7 +39,7 @@ class MetaExpr(object):
 
     """Metadata container for a compute-intensive expression."""
 
-    def __init__(self, type, parent, loops_info, domain):
+    def __init__(self, type, parent, loops_info, domain, mode=0):
         """Initialize the MetaExpr.
```

Comment:
Sure. So the rewrite mode is about an expression. The optimisation level includes various things, among which the rewrite mode level.